app/controllers/exercises_controller.rb

```diff
@@ -16,6 +16,13 @@ def index
   # GET /exercises/1
   # GET /exercises/1.json
   def show
+
+    exercise_relation = ExerciseRelation.find_by(clone_id: @exercise.id)
+    if exercise_relation
+      @exercise_relation = exercise_relation
+      @exercise_relation.origin = exercise_relation.origin
+      @exercise_relation.clone = exercise_relation.clone
```

Comment:
Are these two lines really necessary? shouldn't origin and clone already be copied with the object?